src/main/java/com/learning/mfscreener/service/HistoricalNavService.java

```diff
@@ -125,16 +122,25 @@ String parseNavData(
         return oldSchemeId;
     }
 
+    String readNextNonEmptyLine(BufferedReader br) throws IOException {
```

Comment:
@coderabbitai, can you please rewrite this method to never go to infinite loop.  When it reached end of file stop it.